ctr-nonce-reuse/src/ctr_game.rs

```diff
@@ -0,0 +1,196 @@
+
+            plaintexts_vec: vec![
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "M"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Tom",  "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "Administrator", "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                //br#"{ "Heartbeat": "Alive",         "Since": "1653924358" }"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "M"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Tom",  "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "Administrator", "Name": "Alex", "Surname": "S"}"#.to_vec(),
+            ],
+            ciphertexts_vec: vec![],
+        }
+    }
+
+    pub fn restart(&mut self) {
+        self.rng = rand::thread_rng();
+
+        self.key = self.rng.gen();
+        self.nonce = self.rng.gen();
+        //self.aes_ctr = Aes128Ctr64LE::new(&key.into(), &nonce.into());
+
+        let flag_str = self.create_random_flag();
+        self.flag_plaintext = flag_str.into();
+
+        self.create_ciphertexts(EAVESDROPPED_MSG_COUNT);
+    }
+
+    fn create_random_flag(&mut self) -> String {
+        let left_random_bytes: [u8; 4] = self.rng.gen();
+        let right_random_bytes: [u8; 4] = self.rng.gen();
+        let mut flag_str = String::new();
+
+        write!(
+            flag_str,
+            "FLAG{{{}_CtR_1sNT_SeCC_{}}}",
+            Self::byte_to_hex(&left_random_bytes),
+            Self::byte_to_hex(&right_random_bytes)
+        )
+        .expect("String formatting can not fail");
+
+        flag_str
+    }
+
+    fn create_ciphertexts(&mut self, count: usize) {
+        self.ciphertexts_vec.clear();
+        let mut chosen_plaintext: Vec<_> = self
+            .plaintexts_vec
+            .choose_multiple(&mut self.rng, count)
+            .collect();
+        self.known_plaintext = (*chosen_plaintext.choose(&mut self.rng).expect("a")).clone();
+        chosen_plaintext.push(&self.flag_plaintext);
+        chosen_plaintext.shuffle(&mut self.rng);
+        for plaintext in chosen_plaintext {
+            let ciphertext = self.encrypt_bytes(plaintext);
+            self.ciphertexts_vec.push(ciphertext);
+        }
+    }
+
+    pub fn get_hex_ciphertexts(&self) -> Vec<String> {
+        let mut result = Vec::new();
+
+        for ciphertext in &self.ciphertexts_vec {
+            result.push(Self::byte_to_hex(ciphertext));
+        }
+
+        result
+    }
+
+    pub fn get_known_plaintext(&self) -> String {
+        String::from_utf8_lossy(&self.known_plaintext).to_string()
+    }
+
+    pub fn submit_flag(&self, flag_request: String) -> bool {
+        flag_request.into_bytes().eq(&self.flag_plaintext)
```

Comment:
The `into_bytes` call creates a new vector, making an allocation. We can avoid that by using `as_bytes()`, which creates a reference to the underlying array:
```suggestion
        flag_request.as_bytes() == self.flag_plaintext
```